Sources/Sentry/SentryTransaction.m

```diff
@@ -0,0 +1,90 @@
+#import "SentryTransaction.h"
+#import "NSDate+SentryExtras.h"
+#import "NSDictionary+SentrySanitize.h"
+#import "SentryCurrentDate.h"
+#import "SentryHub.h"
+#import "SentryId.h"
+#import "SentrySpanContext.h"
+#import "SentrySpanId.h"
+#import "SentryTransactionContext.h"
+
+@interface
+SentryTransaction () {
+    SentrySpanContext *trace;
+}
+
+/**
+ * A hub this transaction is attached to.
+ */
+@property (nonatomic, readonly) SentryHub *_Nullable hub;
+
+@end
+
+@implementation SentryTransaction
+
+- (NSDictionary<NSString *, id> *)serialize
+{
+    if (nil == self.timestamp) {
+        self.timestamp = [SentryCurrentDate date];
```

Comment:
`m`: Why do we set the timestamp here? Shouldn't this only happen in `finish`?

Reply:
Good catch. This should definitely be fixed.
@marandaneto is this similar to what we discussed about not compensating when this is `nil`? 
Meaning we send `nil` to the server if it wasn't `finished` properly?

@brustolin Or does the server complain and this was a workaround or something?

Reply:
I copied this behavior from SentryEvent serialization before I start to cascade serialization. I do think this does not belong here, but SentryEvent serialization will do it anyway.

Reply:
This was copied over from event, before transaction inherited from event. Now we don't need it anymore. But SentryEvent will do it anyway, so the timestamp is not only set at finish()

Reply:
Sounds like we should address this. this value should be set in the SentryEvent constructor (for errors) but not for Transactions.

Reply:
yeah we can't do that, as @bruno-garcia said.

a SentryEvent sets a `timestamp` on ctor, when the instance is created (always, consider non-nullable).

a Transaction has a nullable `timestamp` during creation.
a Transaction sets a `timestamp` when `finish` is called (always, consider non-nullable).
ps: if the user forgot to `transaction.finish()`, we should not set a `timestamp` automatically, it should send as it is, same for Spans